common/persistence/cassandraPersistence.go

```diff
@@ -2276,10 +2276,10 @@ func (d *cassandraPersistence) updateActivityInfos(batch *gocql.Batch, activityI
 	}
 }
 
-func (d *cassandraPersistence) updateTimerInfos(batch *gocql.Batch, timerInfos []*TimerInfo, deleteInfos []string,
+func (d *cassandraPersistence) updateTimerInfos(batch *gocql.Batch, timerInfos map[*TimerInfo]struct{}, deleteInfos map[string]struct{},
```

Comment:
why are you changing this to map?

Reply:
according to what i have seen during debugging, the slice can contains duplicates,